### PR TITLE
Added email functionality

### PR DESCRIPTION
**Pull request by NearTruth**

### CODE REVIEW

File: scripts/send_emails.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        except Exeption:
+            pass
+        return users
+
+
+def send_email(username, subject, message):
+    """
+    send email containing the subject and the message to {username}@bcgsc.ca
+
+    message should contain no more than 100 lines of filenames
```
Comment (NearTruth) [inline scripts/send_emails.py:42]: update.  100 lines is incorrect

File: scripts/send_emails.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    karsanlab as a set
+    """
+    
+    if DEV: # DEV mode turns off shell calls and prints emails to stdout
+        return TEST_ACTIVE_USER_SET
+    else:
+        users = set()
+        try:
+            getent_output = subprocess.run(["getent", "group", "karsanlab"], stdout.subprocess.PIPE)
+            users = set(getent_output.stdout.decode().strip().split(":")[3].split(","))
```
Comment (NearTruth) [inline scripts/send_emails.py:30]: nice to have: example output of getent

File: scripts/send_emails.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+"""
+module sends emails to specified receivers
+for use with parse_ncdu_json.py
+"""
+
+import subprocess
+import humanfriendly
+
+MESSAGE_LENGTH = 50
+INACTIVE_USER = "jbridgers"
```
Comment (NearTruth) [inline scripts/send_emails.py:12]: rename variable

File: scripts/parse_ncdu_json.py [hunk 4]
```diff
@@ -72,18 +74,22 @@ def write_summary_table_file(files_list, outfile_name):
...
+                writer.writerow(file[0:1])
+
+
+def write_summary_table_file_users(files_list, outfile_name):
+    """Write file containing a table of file paths and sizes"""
+    with open(outfile_name, 'w') as outfile:
+        writer = csv.writer(outfile, delimiter='\t', lineterminator='\n')
+        writer.writerow(['file_path', 'file_owner'])
+        if files_list:
+            for file in sort_files_list_by_size(files_list):
```
Comment (NearTruth) [inline scripts/parse_ncdu_json.py:86]: review use of "file" likely needs to be changed to something like file_metadata

File: scripts/send_emails.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+def make_dict_files_by_username(file_list):
+    """
+    input format: lst([path, size, username])
+    given list of files, return a dictionary with username as key and a sorted list of files as
+    value
+    output format: dict{username: lst(file)}
+    """
+    user_dict = {}
+    for f in file_list:
```
Comment (NearTruth) [inline scripts/send_emails.py:62]: don't use abbreviations in code

File: scripts/send_emails.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    else:
+        try:  # be careful shell=True is a security risk
+            subprocess.run(f"printf '{message}' | mail -s '{subject}' {username}@bcgsc.ca", shell=True)
+        except Exception:  # handle exceptions
+            pass
+
+
+def make_dict_files_by_username(file_list):
+    """
+    input format: lst([path, size, username])
```
Comment (NearTruth) [inline scripts/send_emails.py:56]: add info on size (type)

File: scripts/send_emails.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+def format_and_send_email(username, sorted_files_list, active_user=True):
+    """
+    format the subject and message then send the email to {username}@bcgsc.ca if the user is an
+    active user, or to {INACTIVE_USER}@bcgsc.ca if not
+
+    sorted_files_list format: lst([file_path, size, uid])
+
+    format files_list into the following way
+    subject line:
+	Files automatically marked for deletion for {_username}
```
Comment (NearTruth) [inline scripts/send_emails.py:91]: change to for review

File: scripts/send_emails.py [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+	Hi {_username},
+	please review and delete unwanted files:
+
+	file1_path (size)
+	file2_path (size)
+	file3_path (size)
+	file4_path (size)
+    """
+
+    subject = f"Files automatically marked for deletion for {username}"
```
Comment (NearTruth) [inline scripts/send_emails.py:103]: change

File: scripts/send_emails.py [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    f"""Hi {username},\n
+please review and delete unwanted files:\n
+"""
+    line_counter = 0
+    message_body = ""
+    for f in sorted_files_list:
+        # print(line_counter)
+        message_body = message_body + f"{f[0]} ({f[1]})\n"
+        line_counter = line_counter + 1
+        if line_counter == 50:  # line_counter == MESSAGE_LENGTH
```
Comment (NearTruth) [inline scripts/send_emails.py:114]: change to message length

File: scripts/send_emails.py [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+            message = message_header + message_body
+            if active_user:
+                send_email(username, subject, message)  # send message of MESSAGE_LENGTH lines
+            else:
+                send_email(INACTIVE_USER, subject, message)
+            message_body = ""
+            line_counter = 0
+
```
Comment (NearTruth) [inline scripts/send_emails.py:122]: add a means for the user to know the file list is truncated

File: scripts/send_emails.py (hunk 8 above)
Comment (NearTruth) [inline scripts/send_emails.py:103]: change

File: scripts/send_emails.py (hunk 3 above)
Comment (jbridgers) [reply scripts/send_emails.py]: variable name does not imply the actual functionality.  Needs updating

File: scripts/send_emails.py (hunk 1 above)
Comment (jbridgers) [reply scripts/send_emails.py]: Needs update

**Comment by jbridgers** [comment]
Please address the code comments

**Pull request merged**